src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -100,11 +100,19 @@ trait SubmissionComponent {
       val query = for {
         submissionRec <- findByWorkspaceId(workspaceContext.workspaceId)
         userRec <- rawlsUserQuery if (submissionRec.submitterId === userRec.userSubjectId)
-      } yield (submissionRec, userRec)
+        methodConfigRec <- methodConfigurationQuery if (submissionRec.methodConfigurationId === methodConfigRec.id)
+        entityRec <- entityQuery if (submissionRec.submissionEntityId === entityRec.id)
+      } yield (submissionRec, userRec, methodConfigRec, entityRec)
       
-      query.result.flatMap(recs => DBIO.sequence(recs.map { case (submissionRec, userRec) =>
-        loadSubmission(submissionRec.id).map(sub => (sub.get, rawlsUserQuery.unmarshalRawlsUser(userRec)))
-      }))
+      query.result.flatMap{recs => DBIO.sequence(recs.map {
```

Comment:
Agreed.  I was able to make it work with `query.result.map` and removal of both `DBIO`s.